### ISSUE TITLE
 Cannot read property 'protectedFields' of undefined with parse server 4.x

### ISSUE THREAD

**Issue by allanext** [parse-community/parse-server#6679]
### Issue Description

I've updated parse server from 3.10.0 to 4.2.0 and without any `protectedFields` configuration i get:

> {"method":"GET","url":"/127-s_domain_com/classes/m_settings","headers":{"host":"127.0.0.1:1337","connection":"keep-alive","content-length":"245","pragma":"no-cache","cache-control":"no-cache","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36","dnt":"1","content-type":"text/plain","accept":"*/*","origin":"http://127.0.0.1:8080","sec-fetch-site":"same-site","sec-fetch-mode":"cors","sec-fetch-dest":"empty","referer":"http://127.0.0.1:8080/login","accept-encoding":"gzip, deflate, br","accept-language":"en-US,en;q=0.9"},"body":{"where":{"deleted_at":null},"limit":1},"level":"verbose","message":"REQUEST for [GET] /127-s_domain_com/classes/m_settings: {\n  \"where\": {\n    \"deleted_at\": null\n  },\n  \"limit\": 1\n}","timestamp":"2020-05-09T17:54:56.792Z"}
> {"message":{},"code":1,"level":"error","stack":"Error: TypeError: Cannot read property 'protectedFields' of undefined\n    at adapter.find.then.catch.error (/Users/alex/dev/domain-parse/node_modules/parse-server/lib/Controllers/DatabaseController.js:1166:21)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:229:7)","timestamp":"2020-05-09T17:54:56.799Z"}

I've tried to add the following configuration:

```
...
publicServerURL: conf.SERVER_URL,
protectedFields: {
      _User: {
        "*": ["email", "phone"]
        //"role:moderator": ["phone"],
        //"role:admin": []
    }
},
liveQuery: {
...
```

but no luck, do i need to configure something else?

### Environment Setup

- **Server**
  - parse-server version 4.2.0
  - Localhost or remote server? localhost

- **Database**
  - MongoDB version: 4.2.3
  - Localhost or remote server? localhost

**Comment by orbitalnine** [comment]
I was getting this error as well. I'm not sure if the cause is the same as in your case but maybe my solution can shed light on the underlying problem.

I was in the process of migrating my parse server setup from AWS to Heroku. I manually exported and imported collections to my new mongodb. I noticed that I was getting this error when querying one collection, but not when querying another. The collection that the error was being thrown on was a collection that didn't seem to be listed in the _SCHEMA table for whatever reason. After manually adding a document for the schema of the missing collection then Parse stopped throwing this error. Not sure if this is helpful at all, but maybe can point towards the problem.
